### PR TITLE
Added auth backend for Astronomer, which allows users to login to Airflow webserver with their Astronomer login

### PR DESCRIPTION
**Pull request by cwurtz**

### CODE REVIEW

File: airflow/contrib/auth/backends/astronomer_auth.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        return self.superuser
+'''
+
+class AstronomerUser(models.User):
+    def __init__(self, user):
+        self.user = user
+
+    @staticmethod
+    def authenticate(username, password):
+        # TODO: make API call to API server
```
Comment (perryao) [inline airflow/contrib/auth/backends/astronomer_auth.py]: @cwurtz good to remove these comments right?

File: airflow/contrib/auth/backends/astronomer_auth.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        # TODO: allow hostname/port/protocol to be configured, set sane default
+        hostname = configuration.get("astronomer-api", "hostname")
+        port = configuration.get("astronomer-api", "port")
+        protocol = configuration.get("astronomer-api", "protocol")
+        endpoint = '%s://%s:%s/v1' % (protocol, hostname, port)
```
Comment (perryao) [inline airflow/contrib/auth/backends/astronomer_auth.py]: @tedmiston should we always prefer .format? Or is that not valid in 2.7+?

Comment (tedmiston) [inline airflow/contrib/auth/backends/astronomer_auth.py]: @perryao @cwurtz `str.format` is valid and preferred over `%` formatting.  I would do:

```python
endpoint = '{}://{}:{}/v1'.format(protocol, hostname, port)
```

I also like building the base URL separately from a particular endpoint for the sake of being DRY for future revisions, unit tests, etc.  So better yet:

```python
from requests.compat import urljoin

base = '{}://{}:{}'.format(protocol, hostname, port)
endpoint = urljoin(base, 'v1')
```

(If you import `requests.compat.urljoin`, it wraps `urllib` for compatibility with Python 2 and 3 vs importing from urllib directly.)

**Comment by cwurtz** [comment]
Fixed the string format and remove comments.  I'll make sure my python code uses .format going forward.

**Comment by tedmiston** [comment]
@cj LGTM.  If you have a chance also check out pycodestyle (formerly pep8) going forward too.  It will catch some things like `'''` instead of `"""` for docstrings.  Usually it's as simple as:

```
(inside your virtual env)
pip install pycodestyle

(inside your project dir)
pycodestyle .
```

Eventually, I'd like to get us to a place where our static analysis and linters are consistent across the board.  I'd like to discuss some of the applicable tools and services at something like a dev brown bag session in the near future (especially since we'll soon have a lot more people writing python @perryao).

**Comment by cwurtz** [comment]
https://github.com/astronomerio/engineering/issues/15

**Pull request merged**